scripts/base/frameworks/logging/main.zeek

```diff
@@ -602,6 +623,109 @@ export {
 	## of the given log record. Note that filter-level policy hooks still get
 	## invoked after the global hook vetoes, but they cannot "un-veto" the write.
 	global log_stream_policy: Log::StreamPolicyHook;
+
+	## Type of function to invoke when delaying a log write has completed.
+	##
+	## This is similar to a :zeek:see:`Log::StreamPolicyHook`, but a callback
+	## of this type is passed to zeek:see:`Log::delay` and executes just before
+	## the record is forwarded to the individual log filters.
+	##
+	## Returning false from a post delay callback discards the log write.
+	type PostDelayCallback: function(rec: any, id: ID): bool;
+
+	## Type of the opaque value returned by :zeek:see:`Log::delay`. These
+	## values can be passed to :zeek:see:`Log::delay_finish` to release a
+	## delayed write operation.
+	type DelayToken: opaque of LogDelayToken;
```

Comment:
I previously had `any` here backed by a `count`. This clashed with the JavaScript integration (it would work with a fixed `count`, just the `any` part would make that icky). But yes, for users it should actually be opaque. Unless we use a random string, users likely use or rely on the value for something and then it's part of the API. Opaques can't be placed in set/tables either which IMO is good here.

Internally we could also attach more to an opaque without needing to worry about script-land consequences in case the need arises.

I'll keep it, unless you have strong opinions.

Reply:
> I'll keep it, unless you have strong opinions.

Yeah this all sounds good to me Arne, thanks.